### PR TITLE
Link environments in Terraform plugin

### LINKED ISSUE #185: Link environments in L0 -- Terraform Plugin
Create new resource, `environment_link`, in Terraform. 
It should look similar to:
```
resource "layer0_environment" "alpha" {
  name = "alpha"
}

resource "layer0_environment" "beta" {
  name = "beta"
}

resource "layer0_environment_link" "alpha_beta" {
  source = "${layer0_environment.alpha.id}"
  dest   = "${layer0_environment.beta.id}"
}
```

### PR DESCRIPTION
**Pull request by tlake**
Closes #185

### CODE REVIEW

File: plugins/terraform/resource_environment_link.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		if strings.Contains(err.Error(), "No environment found") {
+			d.SetId("")
+			log.Printf("[WARN] Error Reading Environment (%s), environment does not exist", sourceID)
+			return nil
+		}
+
+		return err
+	}
+
+	foundLink := false
```
Comment (zpatrick) [inline plugins/terraform/resource_environment_link.go]: [nit] let's shorten these blocks:
```
for _, v := range sourceEnvironment.Links {
       if v == destID {
                return nil
       }
}


d.SetId("")
log.Printf("[WARN] Error Reading Environment Link (%s), link does not exist", sourceID)
return nil
```

File: plugins/terraform/resource_environment_link.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceLayer0EnvironmentLinkCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(client.Client)
+
+	if err := client.CreateLink(d.Get("source").(string), d.Get("dest").(string)); err != nil {
```
Comment (zpatrick) [inline plugins/terraform/resource_environment_link.go]: [super nit] just being consistent with other functions:
```
client := meta.(client.Client)
sourceID := d.Get("source").(string)
destID := d.Get("dest").(string)

if err := client.CreateLink(sourceID, destID); ...
```

Comment (tlake) [inline plugins/terraform/resource_environment_link.go]: ✅

File: plugins/terraform/resource_environment_link.go (hunk 1 above)
Comment (tlake) [inline plugins/terraform/resource_environment_link.go]: ✅

**Pull request merged**